js/main.js

```diff
@@ -317,26 +327,29 @@ function cityIdToLabel(s) {
 /*
  * Set the current city.
  *
- * `city` is the city ID.
+ * `cityID` is the new city's ID. If `newHistoryState` is true then a new
+ * entry in the browser's history is created for the change. Otherwise the
+ * current history entry is replaced.
  */
-function setCity(city) {
-    var filename = 'cities/' + city + '.json';
+function setCity(cityID, newHistoryState) {
```

Comment:
Monadic functions are much easier to understand than dyadic functions. One could not guess what the boolean in `setCity(getHashCity(), false);` does without looking at the function declaration or even at the implementation. Therefore, if possible avoid more the one argument (especially boolean flags). If it is not possible to split the function into two separate functions then the function name should reflect the meaning of the second parameter.